Src/FluentAssertions/EventRaisingExtensions.cs

```diff
@@ -62,33 +62,25 @@ public static IEventRecording WithArgs<T>(this IEventRecording eventRecording, E
 
             Func<T, bool> compiledPredicate = predicate.Compile();
 
-            bool hasArgumentOfRightType = false;
             var eventsMatchingPredicate = new List<OccurredEvent>();
 
             foreach (OccurredEvent @event in eventRecording)
             {
                 var typedParameters = @event.Parameters.OfType<T>().ToArray();
-                if (typedParameters.Any())
-                {
-                    hasArgumentOfRightType = true;
-                }
 
                 if (typedParameters.Any(parameter => compiledPredicate(parameter)))
                 {
                     eventsMatchingPredicate.Add(@event);
                 }
             }
 
-            if (!hasArgumentOfRightType)
-            {
-                throw new ArgumentException("No argument of event " + eventRecording.EventName + " is of type <" + typeof(T) + ">.");
-            }
+            bool hasFoundArgumentsBefore = eventsMatchingPredicate.Any();
 
-            if (!eventsMatchingPredicate.Any())
-            {
-                Execute.Assertion
-                    .FailWith("Expected at least one event with arguments matching {0}, but found none.", predicate.Body);
-            }
+            Execute.Assertion
+                .ForCondition(hasFoundArgumentsBefore)
+                .FailWith($"Expected at least one event with arguments matching either the type <{{0}}> or parameters {{1}}, but found none.",
```

Comment:
Ah?

That don't get into my mind, because a parameterless call of `WithArgs` with just a type is also matching. That is an or in this case, isn't it?

Might be that I am on the wrong track here..

Reply:
That's true as well. So it expects either the type of the argument to match, and optionally, the predicate.

Reply:
ok, got it :) thanks for clarification